javascript-client/src/client.ts

```diff
@@ -49,7 +55,41 @@ export type S3FileFieldProgressCallback = (progress: S3FileFieldProgress) => voi
 export interface S3FileFieldClientOptions {
   readonly baseUrl: string;
   readonly onProgress?: S3FileFieldProgressCallback;
-  readonly apiConfig?: AxiosRequestConfig
+  readonly apiConfig?: AxiosRequestConfig;
+}
+
+function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => {
+    window.setTimeout(() => { resolve(); }, ms);
+  });
+}
+
+function shouldRetry(error: Error): boolean {
+  // We only retry requests under certain failure modes. Namely, either
+  // network errors, or a subset of HTTP error codes.
+  const axiosErr = (error as AxiosError);
+  return axiosErr.isAxiosError && (
+    !axiosErr.response
+    || [429, 500, 502, 503, 504].includes(axiosErr.response.status)
```

Comment:
Why not all `5xx` errors?

Reply:
@danlamanna suggested this list, and I added one extra code to his suggestion. I think I prefer a whitelist of specific codes, such that any "unexpected" situation will break out of our retry loop, but I'm happy to expand that list if there are other codes you want to add.

Reply:
`429` makes sense, but I think we should retry the entire range of `5xx` errors unless we know it's inappropriate. @danlamanna what do you think?